### PR TITLE
Bugfix: decimal separator parsing

### PR DESCRIPTION
**Pull request by WhiteFang5**
## Description

* Fixed parsing of 'stackSize' values that contain a decimal separator
* Changed some of the item properties from textual values to actual numbers
* Updated item templates/displays accordingly to support this change.

## Replication Steps

price-check any of the following items:
```Rarity: Currency
Orb of Alteration
--------
Stack Size: 1,059/20
--------
Reforges a magic item with new random modifiers
--------
Right click this item then left click a magic item to apply it.
Shift click to unstack.
```
```
Rarity: Currency
Orb of Alteration
--------
Stack Size: 1.059/20
--------
Reforges a magic item with new random modifiers
--------
Right click this item then left click a magic item to apply it.
Shift click to unstack.
```
```
Rarity: Currency
Orb of Alteration
--------
Stack Size: 1059/20
--------
Reforges a magic item with new random modifiers
--------
Right click this item then left click a magic item to apply it.
Shift click to unstack.
```
Only some of them work, while others result in a stack size of `1`.  
The expected result would be that they're all considered equal, which is that this PR is fixing.

## How Has This Been Tested?

- [ ] ran `npm run ng:lint`
- [ ] ran `npm run format`
- [ ] ran `npm run ng:test`
- [ ] added unit tests
- [x] manually tested

### CODE REVIEW

**Review by wratho — CHANGES_REQUESTED** [review]

File: src/app/shared/module/poe/service/item/query/item-search-filters-miscs.service.ts [hunk 1]
```diff
@@ -48,8 +48,8 @@ export class ItemSearchFiltersMiscsService implements ItemSearchFiltersService {
 
     if (prop.gemExperience) {
       const splittedExp = prop.gemExperience.value.text.split('/')
-      const exp = +(splittedExp[0] || '').split('.').join('')
-      const expMax = +(splittedExp[1] || '').split('.').join('')
+      const exp = +(splittedExp[0] || '').split("., ").join('')
```
[the remark this answers is not in the document]
Comment (WhiteFang5) [reply src/app/shared/module/poe/service/item/query/item-search-filters-miscs.service.ts:51]: Fixed the regex thing in https://github.com/PoE-Overlay-Community/PoE-Overlay-Community-Fork/pull/180/commits/96e7fb6e1222ecc53c9fb8191ce49022dfc14cd2

File: src/app/shared/module/poe/service/item/parser/item-section-properties-parser.service.ts [hunk 2]
```diff
@@ -104,72 +83,62 @@ export class ItemSectionPropertiesParserService implements ItemSectionParserServ
...
-    props: ItemValueProperty[]
-  ): ItemValueProperty[] {
+  private parseNumber(text: string): number {
+    return +text.split(/[\+%,\. ]+/).join('')
+  }
+
+  private parsePhrase(line: string, phrase: string): [string, boolean] {
     if (line.indexOf(phrase) !== 0) {
-      return props
+      return [undefined, false];
```
[the remark this answers is not in the document]
Comment (WhiteFang5) [reply src/app/shared/module/poe/service/item/parser/item-section-properties-parser.service.ts:134]: The return value is later on split into its variables again (`const [text, augmented] = this.parsePhrase(line, phrase);`), and then `text` if checked if "it's not defined" using `if (!text) return;`. I'm no TypeScript expert, but I believe that if `''` is returned, it the if-statement would evaluate to be false (and this skip the early return)

[the remark this answers is not in the document]
Comment (WhiteFang5) [reply src/app/shared/module/poe/service/item/parser/item-section-properties-parser.service.ts]: Changed to empty string in b2f8b884d7102ca4d8db4b2f44d65e40f5b43db4

File: src/app/shared/module/poe/service/item/query/item-search-filters-miscs.service.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (WhiteFang5) [reply src/app/shared/module/poe/service/item/query/item-search-filters-miscs.service.ts]: Fixed in e08624b6930f4c280b3ab6cb3dc471e1819ad54e

**Review by wratho — APPROVED** [review]

**Pull request merged**